### PR TITLE
[fuchsia] Add a diagnostics directory to the set of remote dirs (#14566) 

### PR DESCRIPTION
**Pull request by miguelfrde**
This reverts commit 4312d37eb1cdd0ed2e7b44647c7156499b741ac7.

 https://fuchsia-review.googlesource.com/c/topaz/+/351729 has landed adding the out/diagnostics dir to all dart components.

### CODE REVIEW

**Comment by miguelfrde** [comment]
@iskakaushik not sure what I did in #15094, got closed and had to re-open again. PTAL

File: shell/platform/fuchsia/flutter/component.cc [hunk 1]
```diff
@@ -233,15 +233,17 @@ Application::Application(
...
         for (auto& dir_str : other_dirs) {
           fidl::InterfaceHandle<fuchsia::io::Directory> dir;
           auto request = dir.NewRequest().TakeChannel();
-          fdio_service_connect_at(directory_ptr_.channel().get(), dir_str,
-                                  request.release());
-          outgoing_dir_->AddEntry(
-              dir_str, std::make_unique<vfs::RemoteDir>(dir.TakeChannel()));
+          auto status = fdio_service_connect_at(directory_ptr_.channel().get(),
+                                                dir_str, request.release());
+          if (status == ZX_OK) {
```
Comment (gaaclarke) [inline shell/platform/fuchsia/flutter/component.cc]: Should you log the error status somehow?

Comment (miguelfrde) [reply shell/platform/fuchsia/flutter/component.cc]: Done

**Comment by gaaclarke** [comment]
Landing on red (more like purple).  The only thing that is broke right now is Fuchsia tests which is flakey.

**Pull request merged**